lib/agenda.js

```diff
@@ -415,108 +550,150 @@ Agenda.prototype.stop = function(cb) {
 
 /**
  * Find and lock jobs
- * @param {String} jobName
- * @param {Object} definition
- * @param {Function} cb
- * @protected
- *  @caller jobQueueFilling() only
+ * @param {String} jobName name of job to try to lock
+ * @param {Object} definition definition used to tell how job is run
+ * @param {Function} cb called when job is locked or errors
+ * @access protected
+ * @caller jobQueueFilling() only
+ * @returns {undefined}
  */
 Agenda.prototype._findAndLockNextJob = function(jobName, definition, cb) {
-  var self = this,
-      now = new Date(),
-      lockDeadline = new Date(Date.now().valueOf() - definition.lockLifetime);
 
-  // Don't try and access Mongo Db if we've lost connection to it. Also see clibu_automation.js db.on.close code. NF 29/04/2015
-  // Trying to resolve crash on Dev PC when it resumes from sleep.
+  var self = this, now = new Date(), lockDeadline = new Date(Date.now().valueOf() - definition.lockLifetime);
+
+  // Don't try and access Mongo Db if we've lost connection to it.
+  // Trying to resolve crash on Dev PC when it resumes from sleep. NOTE: Does this still happen?
```

Comment:
We should confirm this is still an issue. Maybe open an issue about it and ref this commit+line.